packages/terser/src/index.js

```diff
@@ -22,48 +22,65 @@ function log_error(...m) {
   console.error('[terser/index.js]', ...m);
 }
 
-// Peek at the arguments to find any directories declared as inputs
-let argv = process.argv.slice(2);
-// terser_minified.bzl always passes the inputs first,
-// then --output [out], then remaining args
-// We want to keep those remaining ones to pass to terser
-// Avoid a dependency on a library like minimist; keep it simple.
-const outputArgIndex = argv.findIndex((arg) => arg.startsWith('--'));
-
-// We don't want to implement a command-line parser for terser
-// so we invoke its CLI as child processes when a directory is provided, just altering the
-// input/output arguments. See discussion: https://github.com/bazelbuild/rules_nodejs/issues/822
-
-const inputs = argv.slice(0, outputArgIndex);
-const output = argv[outputArgIndex + 1];
-const residual = argv.slice(outputArgIndex + 2);
-
-// user override for the terser binary location. used in testing.
-const TERSER_BINARY = process.env.TERSER_BINARY || require.resolve('terser/bin/uglifyjs')
-// choose a default concurrency of the number of cores -1 but at least 1.
-const TERSER_CONCURENCY = (process.env.TERSER_CONCURRENCY || os.cpus().length - 1) || 1
-
-log_verbose(`Running terser/index.js
-  inputs: ${inputs}
-  output: ${output}
-  residual: ${residual}`);
+/**
+ * @param {string[]} residualArgs
+ * @param {string} output
+ */
+function processSourceMapArg(residualArgs, output) {
+  // copy args so we don't accidently mutate and process the same set of args twice
+  const argsCopy = [...residualArgs];
+  const sourceMapIndex = argsCopy.findIndex(a => a === '--source-map');
```

Comment:
nit:

a little simpler to use indexOf: `const sourceMapIndex = residualArgs.indexOf('--source-map');`

You can also do this check on `residualArgs` and escape with
```
if (sourceMapIndex === -1)
    return residualArgs;
```
and then make a copy of the args after that.